exporter/dynatraceexporter/metrics_exporter.go

```diff
@@ -94,31 +112,141 @@ func (e *exporter) serializeMetrics(md pdata.Metrics) ([]string, int) {
 			metrics := libraryMetric.Metrics()
 			for k := 0; k < metrics.Len(); k++ {
 				metric := metrics.At(k)
-				name, normalizationError := normalizeMetricName(e.cfg.Prefix, metric.Name())
-				if normalizationError != nil {
-					dropped++
-					e.logger.Error(fmt.Sprintf("Failed to normalize metric name: %s", metric.Name()))
-					continue
+				metricLines, err := e.serializeMetric(metric)
+				if err != nil {
+					e.logger.Sugar().Warnf("failed to serialize int gauge %s - %s", metric.Name(), err.Error())
 				}
 
-				var l []string
-				switch metric.DataType() {
-				case pdata.MetricDataTypeNone:
+				if metricLines == nil {
 					continue
-				case pdata.MetricDataTypeGauge:
-					l = serialization.SerializeNumberDataPoints(name, metric.Gauge().DataPoints(), e.cfg.Tags)
-				case pdata.MetricDataTypeSum:
-					l = serialization.SerializeNumberDataPoints(name, metric.Sum().DataPoints(), e.cfg.Tags)
-				case pdata.MetricDataTypeHistogram:
-					l = serialization.SerializeHistogramMetrics(name, metric.Histogram().DataPoints(), e.cfg.Tags)
 				}
-				lines = append(lines, l...)
-				e.logger.Debug(fmt.Sprintf("Exporting type %s, Name: %s, len: %d ", metric.DataType().String(), name, len(l)))
+
+				lines = append(lines, metricLines...)
+				e.logger.Debug(fmt.Sprintf("Serialized %s, Name: %s, lines: %d ", metric.DataType().String(), metric.Name(), len(metricLines)))
 			}
 		}
 	}
+	return lines
+}
 
-	return lines, dropped
+func (e *exporter) dimensions(labels pdata.StringMap) dimensions.NormalizedDimensionList {
```

Comment:
```suggestion
func (e *exporter) makeCombinedDimensions(labels pdata.StringMap) dimensions.NormalizedDimensionList {
```
Might be a bit clearer?